### PR TITLE
pretty improvements

### PR DESCRIPTION
**Pull request by madelynnblue**

### CODE REVIEW

**Comment by mjibson** [comment]
Ok you've convinced me tests are good. I am converted.

File: pkg/sql/sem/tree/testdata/pretty/create_view.align-deindent.golden [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	quotes
+) AS   SELECT startrek.episodes.season, count(*)
+         FROM startrek.quotes
+              JOIN startrek.episodes
+              ON startrek.quotes.episode = startrek.episodes.id
+     GROUP BY startrek.episodes.season
+
+65:
+-----------------------------------------------------------------
+CREATE VIEW startrek.quotes_per_season (season, quotes) AS SELECT
```
Comment (knz) [inline pkg/sql/sem/tree/testdata/pretty/create_view.align-deindent.golden]: I think I'd prefer the AS to come under the VIEW if possible...

**Review by madelynnblue — COMMENTED** [review]
Added partition and interleave. RFAL.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29505#-:-LLcGVtN3hRtalLGrY_P:b-95teg2)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/sql/sem/tree/testdata/pretty/create_view.align-deindent.golden, line 145 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/29505#-LLYn8eGhamQ6juvwDZG-r2-145:-LLcGUgJ8HcvU1s88193:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee661bc5eceb882e2c244d184a1e6fe144697e05/pkg/sql/sem/tree/testdata/pretty/create_view.align-deindent.golden#L145)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I think I'd prefer the AS to come under the VIEW if possible...
</blockquote></details>

Done.

File: pkg/sql/sem/tree/pretty.go [hunk 2]
```diff
@@ -1022,40 +1022,60 @@ func (node *CreateTable) doc(p *PrettyCfg) pretty.Doc {
...
-			d = pretty.ConcatLine(d, p.Doc(node.PartitionBy))
+			docs = append(docs, p.Doc(node.PartitionBy))
 		}
+		d = pretty.Group(pretty.Stack(docs...))
 	}
 	return d
 }
 
+func (node *CreateView) doc(p *PrettyCfg) pretty.Doc {
+	d := pretty.Concat(
```
Comment (knz) [inline pkg/sql/sem/tree/pretty.go]: ConcatSpace and drop the space after VIEW.

**Review by knz — APPROVED** [review]
LGTM with nit

**Pull request merged**